File: frontend/routes/room.route.js
const router = require("express").Router();

const {
  middlewareCheckGoogleProfile,
  middlewareCheckEmail,
} = require('../controllers/oauth-google.controller');

const {
  getSingleRoom,
  putAddMember,
  putExitRoom,
  putRenameRoom,
  putMemberToModerator,
  putModeratorToMember,
  putKickMember,
  postNewChat,
  putEnableNotif,
  putDisableNotif,
  deleteRoom,
} = require('../controllers/room.controller');

router.get(/\/([A-Za-z0-9]+)/, middlewareCheckGoogleProfile, middlewareCheckEmail, getSingleRoom);
router.put(/\/([A-Za-z0-9]+)\/addMember/, middlewareCheckGoogleProfile, middlewareCheckEmail, putAddMember);
router.put(/\/([A-Za-z0-9]+)\/exit/, middlewareCheckGoogleProfile, middlewareCheckEmail, putExitRoom);
router.put(/\/([A-Za-z0-9]+)\/rename/, middlewareCheckGoogleProfile, middlewareCheckEmail, putRenameRoom);
router.put(/\/([A-Za-z0-9]+)\/memberToModerator/, middlewareCheckGoogleProfile, middlewareCheckEmail, putMemberToModerator);
router.put(/\/([A-Za-z0-9]+)\/ModeratorToMember/, middlewareCheckGoogleProfile, middlewareCheckEmail, putModeratorToMember);
router.put(/\/([A-Za-z0-9]+)\/kickMember/, middlewareCheckGoogleProfile, middlewareCheckEmail, putKickMember);
router.post(/\/([A-Za-z0-9]+)\/newChat/, middlewareCheckGoogleProfile, middlewareCheckEmail, postNewChat);
router.put(/\/([A-Za-z0-9]+)\/enableNotif/, middlewareCheckGoogleProfile, middlewareCheckEmail, putEnableNotif);
router.put(/\/([A-Za-z0-9]+)\/disableNotif/, middlewareCheckGoogleProfile, middlewareCheckEmail, putDisableNotif);
router.delete(/\/([A-Za-z0-9]+)\/deleteRoom/, middlewareCheckGoogleProfile, middlewareCheckEmail, deleteRoom);

module.exports = router;

File: backend/routes/room.route.go
package routes

import (
	"github.com/gofiber/fiber/v2"
	controller "github.com/simopunkc/chirpbird-v2/controllers"
)

func RoomRoute(app fiber.Router) {
	app.Get("/:room", controller.GetSingleRoom)
	app.Put("/:room/addMember", controller.PutAddMember)
	app.Put("/:room/exit", controller.PutExitRoom)
	app.Put("/:room/rename", controller.PutRenameRoom)
	app.Put("/:room/memberToModerator", controller.PutMemberToModerator)
	app.Put("/:room/ModeratorToMember", controller.PutModeratorToMember)
	app.Put("/:room/kickMember", controller.PutKickMember)
	app.Post("/:room/newChat", controller.PostNewChat)
	app.Put("/:room/enableNotif", controller.PutEnableNotif)
	app.Put("/:room/disableNotif", controller.PutDisableNotif)
	app.Delete("/:room/deleteRoom", controller.DeleteRoom)
}
